torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -48,7 +182,47 @@ RegisterOperators reg({
         },
         c10::AliasAnalysisKind::INTERNAL_SPECIAL_CASE),
 });
-}
+
+RegisterOperators reg_guard({
+    Operator(
+        prim::CudaFusionGuard,
+        [](const Node* node) -> Operation {
+          return [node](Stack* stack) {
+            FUSER_PERF_SCOPE("CudaFusionGuard");
+            // TODO: check latency here!!!!
+            std::vector<TypePtr> types = node->tys(attr::types);
+            const auto num_inputs = types.size();
+            at::ArrayRef<IValue> inputs = last(stack, num_inputs);
+            drop(stack, num_inputs);
+
+            if (!fuser::cuda::getCudaFusionGuardMode()) {
```

Comment:
in which case do we insert `prim::CudaFusionGuard` but we actually want to skip the check?

Reply:
CI tests. I just want to make sure that there is no accidental check failure that would not hide any codegen error.

But as you pointed this out, I should actually change this to: assert on check success instead of skipping the check